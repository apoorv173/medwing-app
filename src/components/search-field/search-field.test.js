import React from 'react';
import { shallow } from 'enzyme';
import SearchField, { handleChange } from './search-field';
import renderer from 'react-test-renderer';

const event = {
	preventDefault:() => {},
	target: {
		name:'searchBox',
		value:''
	}
};

it('Search onchange event', () => {
	const onChangeEvent = jest.fn(event);
	const searchField = shallow(<SearchField 
			name="searchBox"
			label="Enter location"
			aria-describedby="Search box"
			placeholder="Please enter the location to search"
			onChange={() => onChangeEvent()}
	/>
	);
	expect(searchField.find('InputField').length).toBe(1);

});

it('renders correctly', () => {
	const onChangeEvent = jest.fn();
  const tree = renderer
    .create(<SearchField 
				name="searchBox"
				label="Enter location"
				aria-describedby="Search box"
				placeholder="Please enter the location to search"
				onChange={onChangeEvent}
		/>
		)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('onChange method would be called if timeout exists', () => {
	const onChangeEvent = jest.fn();
	jest.useFakeTimers();
	handleChange(event, onChangeEvent);
	jest.runAllTimers();
	expect(onChangeEvent).toHaveBeenCalled();
	jest.useRealTimers();
});
