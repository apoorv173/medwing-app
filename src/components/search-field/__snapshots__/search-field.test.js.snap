// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="searchBox"
  >
    Enter location
  </label>
  <input
    className="form-control"
    name="searchBox"
    onChange={[Function]}
    placeholder="Please enter the location to search"
    type="text"
  />
</div>
`;
